roles/pulibrary.elastic/tasks/main.yml

```diff
@@ -1,33 +1,48 @@
 ---
-- name: install APT transport for Elasticsearch repository.
+- name: Elastic | install APT transport for Elasticsearch repository.
   apt:
     name: apt-transport-https
     state: present
 
-- name: Add Elasticsearch apt key.
+- name: Elastic | Add Elasticsearch apt key.
   apt_key:
     url: https://artifacts.elastic.co/GPG-KEY-elasticsearch
     state: present
 
-- name: Add Elasticsearch repository.
+- name: Elastic | Add Elasticsearch repository.
   apt_repository:
     repo: 'deb https://artifacts.elastic.co/packages/{{ elasticsearch_version }}/apt stable main'
     state: present
     update_cache: true
 
-- name: Install Elasticsearch.
+- name: Elastic | Modify hosts file
+  template:
+    src: hosts.j2
+    dest: /etc/hosts
+    owner: root
+    group: root
+    mode: 0644
+  when: running_on_server
+
+- name: Elastic | Install Elasticsearch.
   apt:
     name: elasticsearch
     state: present
 
-- name: Configure Elasticsearch.
-  template:
-    src: elasticsearch.yml.j2
-    dest: /etc/elasticsearch/elasticsearch.yml
-    owner: root
-    group: elasticsearch
-    mode: 0660
-  notify: restart elasticsearch
+- name: Elastic | Pre-installation flight check
+  stat:
+    path: /etc/elasticsearch
+  register: elastic_conf_dir
+
+- name: Elastic | disable elastic
+  service:
+    enabled: false
+    masked: true
+    name: elasticsearch
+  when: elastic_conf_dir is not defined
+
+- include_tasks: configure.yml
+- include_tasks: post_configuration.yml
```

Comment:
I don't see a `post_configuration.yml` file in this PR? Is it missing or is this not needed?

Reply:
thanks for the catch. It is missing. It is not needed to go "green" because all the steps on run on the server.